runtime/service.go

```diff
@@ -1201,10 +1206,54 @@ func (s *service) State(requestCtx context.Context, req *taskAPI.StateRequest) (
 	// These fields are pointing files inside the VM.
 	// Replace them with the corresponding files on the host, so clients can access.
 	s.fifosMu.Lock()
+	host := s.fifos[req.ID][req.ExecID]
 	defer s.fifosMu.Unlock()
-	resp.Stdin = s.fifos[req.ID][req.ExecID].Stdin
-	resp.Stdout = s.fifos[req.ID][req.ExecID].Stdout
-	resp.Stderr = s.fifos[req.ID][req.ExecID].Stderr
+
+	if resp.Status != task.StatusRunning {
+		logger.Debug("task is no longer running")
+		return resp, nil
+	}
+
+	resp.Stdin = host.Stdin
+	resp.Stdout = host.Stdout
+	resp.Stderr = host.Stderr
+
+	state, err := s.ioProxyClient.State(requestCtx, &ioproxy.StateRequest{ID: req.ID, ExecID: req.ExecID})
+	if err != nil {
```

Comment:
What's the correct behaviour here? In the event that we have an error, do we not want to create a proxy? Just want to make sure we aren't missing anything.

Reply:
If we cannot call State(), then the chance of having success in the later stage is low, so it would be better to not create a new proxy, I think.

Reply:
What kind of errors do we expect to see here in the event of an error?

Reply:
There is a chance that a process was running when we call`s.agentClient.State(requestCtx, req)`, but exited before calling `s.ioProxyClient.State(requestCtx, &ioproxy.StateRequest{ID: req.ID, ExecID: req.ExecID})`. Then this state call would have codes.NotFound.